tests/phpunit/tests/html-api/wpHtmlProcessor.php

```diff
@@ -182,6 +182,145 @@ public function test_cannot_nest_void_tags( $tag_name ) {
 		);
 	}
 
+	/**
+	 * Test the behavior of expects_closer with regular tags.
+	 *
+	 * @ticket 61257
+	 */
+	public function test_expects_closer_regular_tags() {
+		$processor = WP_HTML_Processor::create_fragment( '<div><p><b><em>' );
+
+		$tags = 0;
+		while ( $processor->next_token() ) {
+			$this->assertTrue(
+				$processor->expects_closer(),
+				"Incorrectly expected a closer on tag {$processor->get_tag()}."
+			);
+			++$tags;
+		}
+
+		$this->assertSame(
+			4,
+			$tags,
+			'Did not find all the expected tags.'
+		);
+	}
+
+	/**
+	 * Test the behavior of with non-tags.
+	 *
+	 * @dataProvider data_non_tags
+	 *
+	 * @param string $html HTML for test.
+	 *
+	 * @ticket 61257
+	 */
+	public function test_expects_closer_non_tags( $html ) {
+		$processor = WP_HTML_Processor::create_fragment( $html );
+
+		$found_token = $processor->next_token();
+
+		if ( WP_HTML_Processor::ERROR_UNSUPPORTED === $processor->get_last_error() ) {
+			$this->markTestSkipped( "HTML {$html} is not supported." );
+		}
+
+		$this->assertTrue( $found_token, "Did not find token in {$html}." );
+
+		$this->assertFalse(
+			$processor->expects_closer(),
+			"Incorrectly expected a closer on void tag {$processor->get_token_type()}."
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public static function data_non_tags() {
+		return array(
+			// We don't pause on doctype?
+			// 'Doctype'                          => array( '<!DOCTYPE html>' ),
```

Comment:
I guess we can drop this?

Reply:
Why?

Reply:
The test was failing because it wasn't finding the `<!DOCTYPE html>` node. I think because HTML ignores that in a document we never pause on it. Maybe that's a bug?